### ISSUE TITLE
Curriculum Template design comps

### ISSUE THREAD

**Issue by hannahkane** [mozilla/teach.mozilla.org#1360]
In #1319 we worked on wireframes for curricular content. Now it's time to turn them into design comps.

**Comment by flukeout** [comment]
I made a couple of preliminary options based on the Kraken teaching kit.

* **Side-nav** - https://redpen.io/cxe30ac854d8460f2f
* **Top Tabs** - https://redpen.io/qcd2e564833084685a

**Comment by chadsansing** [comment]
Very cool, @flukeout - both are giant steps forward. I lean toward the persistent side-bar navigation, but would love to see testing on the tabs, as well.

One additional field to account for: standards. Might be 21st C, might be local, state/provincial, or national.If we can reserve space for us and community contributors to include standards, that would help the template bridge formal and informal spaces and their outcomes. If we could make space for only one, I'd go with 21st C Skills since it supports our work with partners like the Afterschool Alliance. Those standards could be button-ized and occupy a row similar to the competency one. Make sense?

Grateful and excited for this work!

**Comment by jgmac1106** [comment]
These are awesome. I love the top menu (especially if it floats).

We also have to remember that a large majority of curriculum testers asked
for paper versions. There are a few ancient issues floating around about a
print stylesheet with simplified texts and no images.

Not sure if this would influence initial design.

**Comment by flukeout** [comment]
@chadsansing @jgmac1106 Thanks for the feedback, I'll iterate a little bit on the 'tabbed menu' design. I'm worried that it will feel awkward with a lot of steps. Do you guys have a sense of how many steps the average kit has? (cc @hannahkane)

The print styles can be completely independent of the layout, so we'll tackle that as a separate layer on top of these designs.

Chad, can you provide a list or link to the 21st C Skills? Are they the ones that we're currently developing along with the web lit map or is there another set.

**Comment by chadsansing** [comment]
@flukeout - those are the ones we're developing with the Web Lit Map. @anmechung, where is the best place to look for the 21st C Skills?

**Comment by jgmac1106** [comment]
problem solving, communicating, collaborating, creativity. Not sure of the
correct tense of verb nor if it matters since the list isn't parallel
anyways

**Comment by hannahkane** [comment]
Love these. Definitely prefer the scalability of the vertical sidebar. Some modules have up to 9 activities! I can't imagine how that would work on the tabbed version.

Also need to think about what happens if we have one more level of navigation, as in the [expanded privacy kit](https://github.com/mozilla/learning-networks/issues/2460).

**Comment by flukeout** [comment]
@chadsansing @jgmac1106 Here's a WIP of the Kraken teaching kid using the **Side-Nav** layout:

http://localhost:8080/WebLiteracyBasics-I/session01-kraken.html#hunt

**Comment by chadsansing** [comment]
The feeling of relief I had when I first opened the page was palpable. Nice job. So horizontal. Such chunked. Very lack of scroll.

Would it be better to take the times in circles on each step and to frame them in green at the upper-right as we do on the overview page? Thoughts, team?

**Comment by cassiemc** [comment]
@flukeout I like the scalability of side nav too. I'd love it more if it explicitly related to the new side nav that @kristinashu designed. Is it possible for the activity to go widescreen? is @kristinashu working on a UI refresh in parallel to this work?

**Comment by flukeout** [comment]
@cassiemc I'm definitely keeping it nice and light style-wise so that we can coordinate styles easily with @kristinashu's work. Want to make sure we have the right UX & navigation style before I dig into the details.

I kept the activity at a fixed width because I don't see the utility in going widescreen - it seems that it will just make the lines of text too long to read comfortably. Happy to discuss or try both out in a prototype though.

**Comment by flukeout** [comment]
@chadsansing I'll work on coordinating the styles of the time indicators.

A bonus with this design if you're viewing the second step in the activity and send someone that link, it will also open to the second step for them, try it out...

http://flukeout.github.io/webmaker-curriculum/WebLiteracyBasics-I/session01-kraken.html#hunt

**Comment by kristinashu** [comment]
Here's the side nav comps if you need it https://redpen.io/p/jh0a4bc04ea2daf52b although it has different purpose so not sure how similar they should be.

I also like the idea of making it a full width layout. You could use larger fonts and lots of padding on the sides to reduce long text lines. I think making it less boxed in will make it feel more modern.

**Comment by hannahkane** [comment]
@kristinashu to help out with this, too

**Comment by chadsansing** [comment]
cc @zee-moz

**Comment by chadsansing** [comment]
@carotejada shared this over [here](https://github.com/mozilla/learning-networks/issues/278).

![timebar](https://cloud.githubusercontent.com/assets/4128600/10817425/01cef6a0-7e10-11e5-81d1-7848eb050418.png)

Do you think there's a need for something like this in the new design? I feel like the new layout and time get at much of what this does. Maybe something like it appears once on the intro page of an activity?

cc @hannahkane, @flukeout, @kristinashu, @cassiemc

**Comment by hannahkane** [comment]
I think it adds a nice visual element. @chadsansing - do all activities include the same basic building blocks?

**Comment by chadsansing** [comment]
No - after the preparation and introduction steps, the titles of each step change, and the amount of time each takes varies by step (so overall times vary by activity, as well).

**Comment by hannahkane** [comment]
I don't think the varying durations would matter - I'm imagining a dynamic implementation where the author enters specific durations for each section. But if we don't have a finite list of sections, I'm not sure how this would work. 

Thoughts from others?

**Comment by chadsansing** [comment]
Can the author also type in strings for the section labels, just give them a class? Comment-in a blank example for cutting and pasting extra steps?

**Comment by hannahkane** [comment]
Sure, we could make the whole thing dynamic. Just not sure what to do about the icons, though, since there would be a finite set, and it sounds like the section headings are different enough that the icons might not make sense in every context.

**Comment by chadsansing** [comment]
Maybe just a few designs to differentiate between (alone | with others) x (online | on a computer| offline)?

**Comment by hannahkane** [comment]
Ah, I see. Maybe a separate ticket to define a system for that kind of meta-data?

**Comment by hannahkane** [comment]
@flukeout @chadsansing - Do we need *two* layers of navigation? There's the navigation within an activity (e.g. within Kraken the Code, there's 1) intro, 2) hunt, 3) decode), but then there's also navigation between modules (e.g. after Kraken the Code, the user may want to advance to the next activity in the sequence, in this case Ping Kong).

**Comment by chadsansing** [comment]
Yes - at least between activities/lesson plans in the same module/kit.

**Comment by hannahkane** [comment]
I believe the latest designs are here: https://redpen.io/p/npde0940e9123a97e2

**Comment by flukeout** [comment]
Hey guys sorry I've been out of the loop here. I guess I haven't been seeing notifications. I've been making some updates and cleaning up the markup and styles for the design in the repdens. Here is the latest...

http://flukeout.github.io/webmaker-curriculum/WebLiteracyBasics-I/session01-kraken.html#overview

**Here is the latest as a Thimble Project**
https://d157rqmxrxj6ey.cloudfront.net/flukeout/17698/#step-1

Obviously there is a problem with that one, as our placement of the Remix button (or icon on mobile) is conflicting with the layout of the template. Gotta give that some thought if we're going to offer a remixable template.

**Comment by flukeout** [comment]
Looks like we've got a few issues to address
* If we make this into a Remixable thimble project, we need to leave room for the Thimble button
  * Unless we re-do how Thimble does things
* Create a print stylesheet
* Address the white gap between the navigation and content at wider sizes

**Comment by flukeout** [comment]
Here's the mobile version of the curriculum if we decide to move the Remix button to the bottom-right (for mobile only), which might be a good idea...

![image](https://cloud.githubusercontent.com/assets/25212/11827097/8753d522-a33e-11e5-8dac-1ffa7a24407d.png)

**Comment by flukeout** [comment]
**Print styles** are looking pretty good...

![image](https://cloud.githubusercontent.com/assets/25212/11828755/525928f0-a34c-11e5-91fa-07c17d094501.png)

**Comment by chadsansing** [comment]
@flukeout, nice work. The lower-right button makes a lot of mobile sense. I like the clean print styles, as well.

**Comment by cassiemc** [comment]
This is looking great @flukeout

**Comment by flukeout** [comment]
**Still to do**
* Make the desktop and mobile designs behave nicely when there are tons of steps

**Comment by jgmac1106** [comment]
or write curriculum without a ton of steps...........

**Comment by hannahkane** [comment]
+1 to moving Remix button on mobile. I'd be hesitant to make that switch on desktop, but seems like a good solution on mobile. Will likely solve issues for others as well, not just our own curriculum template.

Print styles look good, too.

**Comment by flukeout** [comment]
@jgmac1106 agreed - but since it's meant to be a template, we can't have it break or behave in unexpected ways when it's pushed to the limits. Gotta be defensive.

**Comment by flukeout** [comment]
I've put in a PR to move the Remix button to the bottom right for mobile mode...

https://github.com/mozilla/thimble.webmaker.org/pull/1312

**Comment by hannahkane** [comment]
Hey @flukeout - where are we on this? Trying to figure out when we'll be able to do the work of migrating Web Lit content to the new template.

**Comment by flukeout** [comment]
Hey @hannahkane - we are looking pretty good. Once the latest changes for Thimble go to production (and move the mobile thimble logo to the bottom right) we'll be able to provide a Thimble project as a template for future activities. 

I'll keep you posted on when those changes hit production (sometime today or tomorrow).

**Comment by flukeout** [comment]
@hannahkane Okay, the latest changes have been pushed to production, so the curriculum template should work now as a remixable project. Let's chat in stand about next steps.

**Comment by hannahkane** [comment]
@flukeout. Rad. Thanks! Let's talk tomorrow.

**Comment by hannahkane** [comment]
@flukeout - should we close this ticket, since you opened #1538?

**Comment by flukeout** [comment]
@hannahkane Yeah, good idea. Work continues in #1538
